crates/spicepod/src/component/dataset.rs

```diff
@@ -131,6 +131,28 @@ impl Dataset {
         }
     }
 
+    #[must_use]
+    pub fn acceleration_params(&self) -> Option<HashMap<String, String>> {
+        if let Some(acceleration) = &self.acceleration {
+            if let Some(refresh_interval) = &acceleration.params {
+                return Some(refresh_interval.clone());
+            }
+        }
+
+        None
+    }
+
+    #[must_use]
+    pub fn engine_secret(&self) -> Option<String> {
+        if let Some(acceleration) = &self.acceleration {
+            if let Some(engine_secret) = &acceleration.engine_secret {
```

Comment:
You can just return `acceleration.engine_secret.clone()` instead of unwrapping it and then rewrapping it in an optional.